src/bin/pg_dump/dump_babel_utils.c

```diff
@@ -1574,3 +1574,135 @@ fixCursorForBbfTableData(Archive *fout,
 	*nfields_new = fixCursorForBbfSqlvariantTableData(fout, tbinfo, buf, *nfields,
 													  sqlvar_metadata_pos);
 }
+
+/*
+ * babelfishDumpOpclassHelper - This particular helper function would be helpful
+ * to dump Babelfish operator class created on the top of built-in data types of
+ * the Postgres. For example, Operator class sys.int_numeric for the type int4.
+ * This is needed because of some bug in postgres due to which Operarors and
+ * Support functions are not being dumped correctly for user defined operator
+ * class defined for built-in datatypes.
+ */
+void
+babelfishDumpOpclassHelper(Archive *fout, const OpclassInfo *opcinfo, PQExpBuffer *buff, bool *needComma)
+{
+	PQExpBuffer	query;
+	PGresult	*res;
+	int			i_opfnamespace;
+	int			i_opfname;
+	char		*opfnamespace;
+	char		*opfname;
+	const char	*str;
+	const char 	*opclass = fmtQualifiedDumpable(opcinfo);
+
+
+	/* Firstly check that Operator class is part of Babelfish */
+	if (strcasecmp(opclass, "\"sys\".\"int_numeric\"") != 0 &&
+		strcasecmp(opclass, "\"sys\".\"numeric_int\"") != 0 &&
+		strcasecmp(opclass, "\"sys\".\"int2_numeric\"") != 0 &&
+		strcasecmp(opclass, "\"sys\".\"numeric_int2\"") != 0 &&
+		strcasecmp(opclass, "\"sys\".\"int8_numeric\"") != 0 &&
+		strcasecmp(opclass, "\"sys\".\"numeric_int8\"") != 0)
+		return;
+
+	query = createPQExpBuffer();
+
+	/* Get the details of Operator family */
+	appendPQExpBuffer(query, "SELECT opfnamespace::regnamespace, "
+					  "opfname "
+					  "FROM pg_opclass c JOIN  pg_opfamily f ON c.opcfamily = f.oid "
+					  "WHERE c.oid = '%u'::pg_catalog.oid",
+					  opcinfo->dobj.catId.oid);
+
+	res = ExecuteSqlQueryForSingleRow(fout, query->data);
+
+	i_opfnamespace = PQfnumber(res, "opfnamespace");
+	i_opfname = PQfnumber(res, "opfname");
+
+	opfnamespace = PQgetvalue(res, 0, i_opfnamespace);
+	opfname = PQgetvalue(res, 0, i_opfname);
+
+	/* Final checks that Operator class is part of built-in operator family. */
+	if (opfnamespace == NULL || opfname == NULL)
+	{
+		PQclear(res);
+		return;
+	}
+
+	if (strcasecmp(opfnamespace, "\"pg_catalog\"") != 0 ||
+		strcasecmp(opfname, "integer_ops") != 0)
+	{
+		PQclear(res);
+		return;
+	}
+
+	PQclear(res);
+	destroyPQExpBuffer(query);
+
+	if (strcasecmp(opclass, "\"sys\".\"int_numeric\"") == 0)
+	{
```

Comment:
(Just a thought): Can't we somehow modify the original query in `dumpOpclass` function so that it automatically dumps these operator classes which we want to dump? We have already done this in past, for example see `fixCursorForBbfTableData` function.

Reply:
Since pg is not registeting dependency between pg_amop and pg_opclass, its impossible to find member functions/operators using any query.

Reply:
Hmm, the only problem with this approach is that any change to opclass definition would require pg_dump utility changes.

Reply:
yes! that is true.

Reply:
We are planning something to bring up this issue with pg community